docs: Add setup documentation for Oasis CLI for all platforms (#541)
oasisprotocol/cli: docs/README.md

Oasis command-line interface (CLI) is a powerful all-in-one tool for
interacting with the Oasis Network. You can download the latest release

> binaries from the [GitHub repository] (see the “Installing or Updating Oasis
> CLI on macOS” section below for macOS-specific instructions).

Reviewer: Let's use ASCII in the markdown sources wherever possible. The HTML layer can later do more fancy rendering.

```suggestion
binaries from the [GitHub repository] (see the "Installing or Updating Oasis
CLI on macOS" section below for macOS-specific instructions).
```

  - transaction encryption with X25519-Deoxys-II envelope
  - support for Ed25519, Ethereum-compatible Secp256k1 and Sr25519 signature
    schemes

>   - raw, BIP-44, ADR-8 and Ledger’s legacy derivation paths

Reviewer: Same.

```suggestion
  - raw, BIP-44, ADR-8 and Ledger's legacy derivation paths
```

  - inspection of blocks, transactions, results and events

[GitHub repository]: https://github.com/oasisprotocol/cli/releases

> ## Installing or Updating Oasis CLI on macOS

Author: sure, porting the section to setup.md and will refer it at the top of https://docs.oasis.io/general/manage-tokens/cli/

  - inspection of blocks, transactions, results and events

[GitHub repository]: https://github.com/oasisprotocol/cli/releases

## Installing or Updating Oasis CLI on macOS

This guide covers the installation or update process for the Oasis CLI on
macOS (compatible with Apple Silicon like M1/M2/M3). It assumes you have
Terminal access and basic command-line knowledge.

### Prerequisites

- macOS (tested on Apple Silicon).
- To avoid modifying system directories, we'll install the binary in a
  user-specific path (`~/.local/bin`). Ensure `~/.local/bin` is in your `PATH`.
  If not, add it by editing your shell configuration file (e.g., `~/.zshrc`
  for zsh):

>   ```bash

Reviewer: These commands are not bash-specific, so `shell` should work just fine. Same below.

````suggestion
  ```shell
````

  echo 'export PATH="$HOME/.local/bin:$PATH"' >> ~/.zshrc
  source ~/.zshrc
  ```

  Create the directory if it doesn't exist:

  ```bash
  mkdir -p ~/.local/bin
  ```

### Installation Steps

1. **Download the Binary**

   - Visit the Oasis CLI [GitHub repository].
   - Download the latest macOS archive, e.g.,
     `oasis_cli_X.Y.Z_darwin_all.tar.gz` (replace `X.Y.Z` with the version,
     like `0.14.1`).

2. **Extract the Archive**

   ```bash
   cd ~/Downloads

>    tar -xzf oasis_cli_0.14.1_darwin_all.tar.gz   # replace with your version

Reviewer: Perhaps use X.Y.Z throughout the doc.

```suggestion
   tar -xzf oasis_cli_X.Y.Z_darwin_all.tar.gz   # replace with your version
```

   ```

>    This creates a directory such as `oasis_cli_0.14.1_darwin_all`.

Reviewer: ```suggestion
   This creates a directory such as `oasis_cli_X.Y.Z_darwin_all`.
```

3. **Navigate to the Extracted Directory**

   ```bash

>    cd oasis_cli_0.14.1_darwin_all

Reviewer: ```suggestion
   cd oasis_cli_X.Y.Z_darwin_all
```

   ```

4. **Move the Binary to the User Path**

   ```bash
   mv oasis ~/.local/bin/
   ```

5. **Bypass macOS Gatekeeper** (if you encounter a security warning)

   ```bash
   xattr -d com.apple.quarantine ~/.local/bin/oasis
   ```

   If a dialog appears, go to **System Settings → Privacy & Security →

>    Security**  
>    and click **“Open Anyway.”**

Reviewer: ```suggestion
   Security** and click **Open Anyway.**
```
